modules/oe_content_event/tests/src/Kernel/EventLinkSourceFilterTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_content_event\Kernel;
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\node\Entity\Node;
+
+/**
+ * Tests the internal link source filters related to Events.
+ *
+ * @covers \Drupal\oe_content_event\Plugin\InternalLinkSourceFilter\EventLinkSourceFilter
+ */
+class EventLinkSourceFilterTest extends EventKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'oe_link_lists',
+    'oe_link_lists_internal_source',
+  ];
+
+  /**
+   * Test that the event filter works as intended.
+   */
+  public function testEventLinkSourceFilter(): void {
+
+    // Create an event that ended in the far past.
+    $values = [
+      'type' => 'oe_event',
+      'title' => 'My node title',
+      'oe_event_dates' => [
+        'value' => '1999-05-10T12:00:00',
+        'end_value' => '1999-05-15T12:00:00',
+      ],
+    ];
+    $ancient_event = Node::create($values);
+    $ancient_event->save();
+
+    // Create an event that ended in the near past.
+    $values = [
+      'type' => 'oe_event',
+      'title' => 'My node title',
+      'oe_event_dates' => [
+        'value' => '2016-05-10T12:00:00',
+        'end_value' => '2016-05-15T12:00:00',
+      ],
+    ];
+    $past_event = Node::create($values);
+    $past_event->save();
+
+    // Create an event starts on the near future.
+    $values = [
+      'type' => 'oe_event',
+      'title' => 'My node title',
+      'oe_event_dates' => [
+        'value' => '2050-05-10T12:00:00',
+        'end_value' => '2050-05-15T12:00:00',
+      ],
+    ];
+    $upcoming_event = Node::create($values);
+    $upcoming_event->save();
+
+    // Create an event starts on the far future.
+    $values = [
+      'type' => 'oe_event',
+      'title' => 'My node title',
+      'oe_event_dates' => [
+        'value' => '2200-05-10T12:00:00',
+        'end_value' => '2200-05-15T12:00:00',
+      ],
+    ];
+    $future_event = Node::create($values);
+    $future_event->save();
+
+    /** @var \Drupal\oe_link_lists_internal_source\InternalLinkSourceFilterPluginManager $plugin_manager */
+    $plugin_manager = $this->container->get('plugin.manager.oe_link_lists.internal_source_filter');
+    /** @var \Drupal\oe_link_lists_internal_source\InternalLinkSourceFilterInterface $plugin */
+    $plugin = $plugin_manager->createInstance('event_link_source_filter', []);
+
+    // Plugin applies to events.
+    $this->assertTrue(TRUE, $plugin->isApplicable('node', 'oe_event'));
+    // Plugin does not apply to news.
+    $this->assertTrue(TRUE, $plugin->isApplicable('node', 'oe_news'));
```

Comment:
I suppose it is should be:
```suggestion
    $this->assertFalse($plugin->isApplicable('node', 'oe_news'));
```

Reply:
Fixed